### PR TITLE
002 add cli tests

### PR DESCRIPTION
**Pull request by mikeerickson**
- added cli tests
- added mocha and chai as dev dependencies
- added CHANGELOG.md
- changed `npm test` package.json script (was defined as `npm run lint`)
- sanity checked with `npm run lint` which uses `eslint`

### CODE REVIEW

**Comment by rickycodes** [comment]
@mikeerickson maybe test should be `npm run lint && mocha` that way both lint and tests run

it also looks like a dependency is missing, hence the failed build on travis.

**Comment by mikeerickson** [comment]
Semantics, but I prefer test to test and lint to lint. 

Could add another key

"all": "npm test && npm run eslint"

Surely not a big deal either way, I usually use bash scripts anyway

alias l="npm run lint"
and "t" from npm shortcut.

**Comment by mikeerickson** [comment]
I only added two dependencies afaik? 

- mocha 
- chai 

What error does Travis report?

**Comment by rickycodes** [comment]
@mikeerickson 

https://travis-ci.org/npms-io/npms-cli/jobs/158306299
`Error: Cannot find module 'lodash'`

**Comment by rickycodes** [comment]
@mikeerickson re: having lint run with `npm test`; this is handy to enforce eslint rules on future pull requests (which is rather handy). If the lint task isn't rolled in that part of things isn't being automated.

**Comment by mikeerickson** [comment]
Hm, don't recall using Lodash, I will have a look.

**Comment by mikeerickson** [comment]
@rickycodes alright, I will update `npm test` to restore linting.

**Comment by mikeerickson** [comment]
@rickycodes seems there is an issue running tests with node 4.5.x, thus is causing travis to fail.  I am looking at this right now

**Comment by mikeerickson** [comment]
@rickycodes OK everything is now green and travis is happy.

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+## angular-sandbox Changelog
```
Comment (satazor) [inline CHANGELOG.md]: Typo on the project name

File: test/info.spec.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+describe(chalk.cyan('==> cmd/info.js'), () => {
+    it('should return info for searched package', (done) => {
+        childProcess.exec(`node ${cmd} info npms-cli --size 1 --output json`, 'utf8', (err, stdout, stderr) => {
+            if (err) { throw err }
+            const resultObj = JSON.parse(stdout)
+
+            assert(resultObj.collected.metadata.name === 'npms-cli');
+        })
+        done()
```
Comment (satazor) [inline test/info.spec.js]: Is this done() correct here? I think this should be promisifed or use buffered-spawn module

File: test/cli.spec.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
+'use strict';
```
Comment (satazor) [inline test/cli.spec.js]: There are some lint erros (commas missing among other things)

File: test/info.spec.js (hunk 2 above)
Comment (satazor) [inline test/info.spec.js]: Actually all done() calls around cp.exec shouldnt be called inside? Alternatively promisify or use execSync

File: test/cli.spec.js (hunk 3 above)
Comment (satazor) [inline test/cli.spec.js]: Please do not disable linting rules on top of the file.. Respect the coding standard defined in the project

File: CHANGELOG.md (hunk 1 above)
Comment (flesch) [inline CHANGELOG.md]: Do we need this file? It's kind of redundant if we use GH releases.

File: test/info.spec.js (hunk 2 above)
Comment (mikeerickson) [inline test/info.spec.js]: @sataroz I originally had them in cp block but was timing out w/ node 4.5.  I realize now it is a bad test as done should indeed be inside. 

I am going to refactor and use the above suggested approach.

File: test/cli.spec.js (hunk 3 above)
Comment (mikeerickson) [inline test/cli.spec.js]: @sataroz my apologies, it was copy pasted and I failed to remove.

File: CHANGELOG.md (hunk 1 above)
Comment (mikeerickson) [inline CHANGELOG.md]: @flesch just a force of habit.  But I do agree it is redundant, will be removed when I address a few other issues @sataroz mentioned

**Comment by satazor** [comment]
See #16

**Pull request closed without merging**